brozzler/frontier.py

```diff
@@ -138,7 +138,13 @@ def claim_sites(self, n=1):
                         emit=lambda acc, site, new_acc: r.branch(
                             r.and_(
                                 r.or_(
-                                    site["claimed"].not_(),
+                                    r.and_(
+                                        site["claimed"].not_(),
+                                        r.or_(
+                                            site.has_fields("last_disclaimed").not_(),
+                                            site["last_disclaimed"].lt(r.now().sub(20)),
```

Comment:
Maybe a better way to do this, but at least comment the intent. We are ensuring a site cannot be reclaimed within 20s of it having been disclaimed to avoid the tight retry loop when the only remaining pages have a retry_after set in the future.

Reply:
I would suggest assigning this to a constant to at least make it semantically clearer. Or maybe put a little more detail in the comment: "We are ensuring a site cannot be reclaimed within 20s of it having been disclaimed to avoid the tight retry loop when the only remaining pages have a retry_after set in the future." would be sufficient.